python/ray/tune/trial_runner.py

```diff
@@ -932,31 +935,6 @@ def _stop_experiment_if_needed(self):
                 if t.status is not Trial.ERROR
             ]
 
-    def _get_next_trial(self):
-        """Replenishes queue.
-
-        Blocks if all trials queued have finished, but search algorithm is
-        still not finished.
-        """
-        no_trials_unfinished = True
-        no_trials_pending = True
-        for trial in self._live_trials:
-            if not trial.is_finished():
-                no_trials_unfinished = False
-            if trial.status == Trial.PENDING:
-                no_trials_pending = False
-            if not no_trials_unfinished and not no_trials_pending:
-                break
-        wait_for_trial = no_trials_unfinished and not self._search_alg.is_finished()
-        # Only fetch a new trial if we have no pending trial
-        if wait_for_trial or no_trials_pending:
```

Comment:
if all trials are finished and search algo is finished, in the old logic, we would enter this branch. But I think it's unnecessary, as `_update_trial_queue` will just return False with no trial added to the queue and thus no trial selected by the `scheduler_alg`.